### ISSUE TITLE
Abs(x)^2 should not simplify to x^2

### ISSUE THREAD

**Issue by uazu** [grzegorzmazur/yacas#331]
This is incorrect for complex numbers:

```
In> (Abs(x))^2
Out> x^2
In> (Abs(3+4*I))^2
Out> 25
In> (3+4*I)^2
Out> Complex(-7,24)
```

**Comment by grzegorzmazur** [comment]
You are absolutely right of course. The problem is that yacas is not very consistent with assuming either real or complex values during calculations. This has to be fixed, but it's a huge undertaking in general.
